src/cli/tls_client.cpp

```diff
@@ -386,53 +387,50 @@ class TLS_Client final : public Command {
       }
 
    private:
+      // Use Socket_Utils types for socket operations
+      using socket_type = Socket_Utils::socket_type;
+      using socket_op_ret_type = Socket_Utils::socket_op_ret_type;
+      using socklen_type = Socket_Utils::socklen_type;
+      using sendrecv_len_type = Socket_Utils::sendrecv_len_type;
+
       static socket_type connect_to_host(const std::string& host, uint16_t port, bool tcp) {
          addrinfo hints;
          std::memset(&hints, 0, sizeof(hints));
          hints.ai_family = AF_UNSPEC;
          hints.ai_socktype = tcp ? SOCK_STREAM : SOCK_DGRAM;
-         addrinfo *res, *rp = nullptr;
+         addrinfo* res = nullptr;
```

Comment:
Similarly to the comments in `socket_udp.cpp`, please wrap `res` into a `unique_ptr` with custom deleter.